### PR TITLE
setup.cfg can also be a place to store the version

### PR DESCRIPTION
**Pull request by ewjoachim**
This is another implementation of #230. I'm not really satisfied because this is more or less copy/pasted code. (If we decide to go this way, I'll integrate the docs in this PR too).

Let me know what you prefer :)

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
Both approaches look okay at first glance. This one seems simpler, with less changes. I have not tried it out. Apparently `setup.cfg` could also contain `version = attr: src.VERSION`. I don't know if that maybe trips up the logic here.
I don't really have time now for a good look, and will be away for a week. And I think @reinout is on vacation but could be back soon.

**Comment by ewjoachim** [comment]
After re-reading myself, this version needs the change in the regex pattern too. This regex pattern includes a digit so we're safe for version = attrs:

I'm totally ok if we leave this PR sleep for a few weeks :) Enjoy your time off !

**Comment by ewjoachim** [comment]
Ping @reinout ?

**Comment by reinout** [comment]
This one seems simpler, so I prefer it.
But: I haven't got experience with version-in-setup.cfg . Does that have to do with the new non-setup.py initiatives?

**Comment by ewjoachim** [comment]
Well, I'm not sure what the new non-setup.py initiatives are, but yes, the idea is mainly to put things that should be static configuration in static config files instead of code files.

For an example of how that goes, see Raincoat's [setup.py](https://github.com/novafloss/raincoat/blob/master/setup.py) and [setup.cfg](https://github.com/novafloss/raincoat/blob/master/setup.cfg). All of this is being made possible by setuptool's new (last december) [ability]((https://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files)) to derive all parameters (sadly, except entrypoint) from setup.cfg.

I'll add the docs from the other PR in this PR.

**Comment by ewjoachim** [comment]
Ok, I've updated and rebased the code and docs. I feel like it's missing tests (well, I know it's missing tests, because I tried it on my project and it didn't work at first). I'd be happy to write it, but the existing tests are in a format that, while readable, is quite new to me, and I'm not sure where and how it would be the most appropriate for me to write those.

**Comment by ewjoachim** [comment]
Ok, it's updated with tests for the new feature. Ready for review !

File: doc/source/versions.rst [hunk 1]
```diff
@@ -21,6 +21,12 @@ version number can come from three different locations:
         version='1.0',
         name='...
 
+- The ``setup.cfg`` file. zest.releaser will look for something like:
```
Comment (reinout) [inline doc/source/versions.rst]: Double colon instead of single one

File: zest/releaser/tests/vcs.txt [hunk 2]
```diff
@@ -245,3 +245,18 @@ Setup.py with an uppercase VERSION:
...
     setup(name='urgh', version=VERSION)
+
+version in setup.cfg:
+
+    >>> setupcfgproject = os.path.join(tempdir, 'setupcfg')
+    >>> os.mkdir(setupcfgproject)
+    >>> os.chdir(setupcfgproject)
+    >>> lines = [
+    ...     "[metadata]",
+    ...     "  version = 1.0"]
```
Comment (reinout) [inline zest/releaser/tests/vcs.txt]: Remove the spaces in front of "version", this breaks the configparser on python 2.7 and on pypy. (That is where the test failure on this PR comes from).

File: CHANGES.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Changelog for zest.releaser
 6.12.4 (unreleased)
 -------------------
 
-- Nothing changed yet.
+- Write version in ``setup.cfg``. [ewjoachim]
```
Comment (reinout) [inline CHANGES.rst]: Oh, and change this in "Also support version in ``setup.cfg``. "

**Comment by reinout** [comment]
Looks good. I've got a few small comments (including a fix for the test failure).

**Comment by ewjoachim** [comment]
All fixed. As it was simple fixes, I've taken the step of squashing fixups directly.

Thank you again !

File: CHANGES.rst [hunk 4]
```diff
@@ -5,6 +5,8 @@ Changelog for zest.releaser
 -------------------
 
 - Nothing changed yet.
+- Also support version in setup.cfg. [ewjoachim]
+- Write version in ``setup.cfg``. [ewjoachim]
```
Comment (ewjoachim) [inline CHANGES.rst]: hm, re-doing this.

Comment (ewjoachim) [reply CHANGES.rst]: done

**Comment by reinout** [comment]
Thanks! I just released 6.12.4, can you check if it all works for you?

**Comment by ewjoachim** [comment]
Works great ! On both usual projects and setup.cfg projects. Thank you !

**Pull request merged**